src/wizards/tokenSwapDealWizard/components/tokenDetails/tokenDetails.ts

```diff
@@ -0,0 +1,122 @@
+import { bindable } from "aurelia-typed-observable-plugin";
+import "./tokenDetails.scss";
+import { IToken } from "../../../../entities/DealRegistrationTokenSwap";
+import { autoinject, bindingMode } from "aurelia-framework";
+import { AureliaHelperService } from "../../../../services/AureliaHelperService";
+import { TokenService } from "../../../../services/TokenService";
+import {
+  validateTrigger,
+  ValidationController,
+  ValidationControllerFactory,
+  ValidationRules,
+} from "aurelia-validation";
+import { PrimeRenderer } from "../../../../resources/elements/primeDesignSystem/validation/primeRenderer";
+import { Validation } from "../../../../services/ValidationService";
+import { Utils } from "../../../../services/utils";
+
+@autoinject
+export class TokenDetails {
+  @bindable token: IToken;
+  @bindable.string() viewMode: "edit" | "view" = "edit";
+  @bindable({defaultBindingMode: bindingMode.fromView}) onDelete: () => void;
+  @bindable({defaultBindingMode: bindingMode.fromView}) form: ValidationController;
+
+  tokenInfoLoading = false;
+  infoSearched = false;
+  saving = false;
+
+  constructor(
+    private aureliaHelperService: AureliaHelperService,
+    private tokenService: TokenService,
+    private validationControllerFactory: ValidationControllerFactory,
+  ) {
+
+    this.form = this.validationControllerFactory.createForCurrentScope();
+    this.form.validateTrigger = validateTrigger.change;
+    this.form.addRenderer(new PrimeRenderer);
+  }
+
+  attached() {
+    ValidationRules.customRule(
+      "isValidIERC20Address",
```

Comment:
Maybe we can add this to the ValidationService so is reusable?  No big deal if is not anticipated to be needed again

Reply:
Yes. I wanted to do that but this validator uses the `this.tokenService` which I don't have access to in the `ValidationService` 😞

Reply:
All you have to do is inject TokenService in ValidationService, no?